### PR TITLE
[infra] Update temple test to allow templefiles without auth services

### PR DESCRIPTION
**Pull request by Lewiky**
The way temple-test had been written it was always assumed an auth service existed, so would try and get an auth token before starting, now we skip that logic if there's no auth service generated

### CODE REVIEW

**Comment by Lewiky** [comment]
Closes #290

**Review by jaylees14 — APPROVED** [review]
Nice catch 😄

File: src/main/scala/temple/test/ProjectTester.scala [hunk 1]
```diff
@@ -27,23 +27,21 @@ object ProjectTester {
...
-        } catch {
-          case _: IOException => // Keep trying
+      try {
+        configuredKong =
+          if (usesAuth)
+            Http("http://localhost:8000/api/auth/login")
+              .method("POST")
+              .asString
+              .code == HttpURLConnection.HTTP_BAD_REQUEST
+          // TODO: this might not be true, needs investigating
```
Comment (c-harding) [inline src/main/scala/temple/test/ProjectTester.scala:37]: has this been investigated yet? If not, whose job is it?

File: src/main/scala/temple/test/ProjectTester.scala [hunk 2]
```diff
@@ -27,23 +27,21 @@ object ProjectTester {
         s"KONG_ADMIN=${config.kongAdminURL} KONG_ENTRY=${config.baseIP} sh $generatedPath/kong/configure-kong.sh",
```
Comment (xsanda) [inline src/main/scala/temple/test/ProjectTester.scala:27]: Should this be in the while loop?

File: src/main/scala/temple/test/ProjectTester.scala [hunk 3]
```diff
@@ -27,23 +27,21 @@ object ProjectTester {
...
-          case _: IOException => // Keep trying
+      try {
+        configuredKong =
+          if (usesAuth)
+            Http("http://localhost:8000/api/auth/login")
+              .method("POST")
+              .asString
+              .code == HttpURLConnection.HTTP_BAD_REQUEST
+          // TODO: this might not be true, needs investigating
+          else true
```
Comment (xsanda) [inline src/main/scala/temple/test/ProjectTester.scala:38]: so if we don’t use auth, we still exec, but we don’t wait to check its consequences?

Comment (xsanda) [inline src/main/scala/temple/test/ProjectTester.scala:38]: so if we don’t use auth, we still exec, but we don’t wait to check its consequences?

**Review by c-harding — COMMENTED** [review]
now I look at this I’m not sure it’s your fault, but there seem to be some weirdness around the kong configuration

**Review by c-harding — APPROVED** [review]
Looks good to me

**Pull request merged**